src/components/external-link/external-link.tsx

```diff
@@ -1,32 +1,37 @@
-import { ReactNode, useEffect } from 'react';
+import { ComponentPropsWithoutRef, useEffect } from 'react';
 
-interface Props {
-  className?: string;
+interface Props extends ComponentPropsWithoutRef<'a'> {
```

Comment:
One last thing, we want to omit the `target` or `rel` props (because we don't want to be able to change that for an ExternalLink):